tests/runtest.py

```diff
@@ -44,11 +90,283 @@
+        core_run = os.path.join(self.core_root, "corerun")
+
+        env = {
+            "COMPlus_AssertOnNYI": "1",
+            "COMPlus_ContinueOnAssert": "0"
+        }
+
+        if self.env is not None:
+            # Convert self.env to a defaultdict
+            self.env = defaultdict(lambda: None, self.env)
+            for key, value in env.iteritems():
+                self.env[key] = value
+            
+        else:
+            self.env = env
+
+        environment = []
+        for key, value in self.env.iteritems():
+            env = {
+                "name": key,
+                "value": value
+            }
+
+            environment.append(env)
+
+        configuration = defaultdict(lambda: None, {
+            "name": self.unique_name,
+            "type": dbg_type,
+            "request": "launch",
+            "program": core_run,
+            "args": [self.exe_location],
+            "stopAtEntry": False,
+            "cwd": os.path.join("${workspaceFolder}", "..", ".."),
+            "environment": environment,
+            "externalConsole": True
+        })
+
+        if self.build_type.lower() != "release":
+            symbol_path = os.path.join(self.core_root, "PDB")
+            configuration["symbolSearchPath"] = symbol_path
+
+        # Update configuration if it already exists.
+        config_exists = False
+        for index, config in enumerate(configurations):
+            if config["name"] == self.unique_name:
+                configurations[index] = configuration
+                config_exists = True
+
+        if not config_exists:
+            configurations.append(configuration)
+        json_str = json.dumps(launch_json,
+                              indent=4, 
+                              separators=(',', ': '))
+
+        with open(launch_json_location, 'w') as file_handle:
+            file_handle.write(json_str)
+
+    def __create_repro_wrapper__(self):
+        """ Create the repro wrapper
+        """
+
+        if self.host_os == "Windows_NT":
+            self.__create_batch_wrapper__()
+        else:
+            self.__create_bash_wrapper__()
+
+    def __create_batch_wrapper__(self):
+        """ Create a windows batch wrapper
+        """
+    
+        wrapper = \
+"""@echo off
+REM ============================================================================
+REM Repro environment for %s
+REM 
+REM Notes:
+REM 
+REM This wrapper is automatically generated by runtest.py. It includes the
+REM necessary environment to reproduce a failure that occured during running
+REM the tests.
+REM
+REM In order to change how this wrapper is generated, see
+REM runtest.py:__create_batch_wrapper__(). Please note that it is possible
+REM to recreate this file by running tests/runtest.py --analyze_results_only
+REM with the appropriate environment set and the correct arch and build_type
+REM passed.
+REM
+REM ============================================================================
```

Comment:
This looks promising! It'll be great if this makes it easy to repro failures. Could you add some documentation on how this would be used?

Reply:
I will address this as a separate PR